homeassistant/components/reolink/__init__.py

```diff
@@ -75,23 +77,42 @@ async def async_device_config_update():
         async with async_timeout.timeout(host.api.timeout):
             await host.renew()
 
-    coordinator_device_config_update = DataUpdateCoordinator(
+    async def async_check_firmware_update():
+        """Check for firmware updates."""
+        async with async_timeout.timeout(host.api.timeout):
+            try:
+                return await host.api.check_new_firmware()
+            except ReolinkError as err:
+                raise UpdateFailed(
+                    f"Error checking Reolink firmware update {host.api.nvr_name}"
+                ) from err
+
+    device_coordinator = DataUpdateCoordinator(
         hass,
         _LOGGER,
         name=f"reolink.{host.api.nvr_name}",
         update_method=async_device_config_update,
-        update_interval=timedelta(seconds=DEVICE_UPDATE_INTERVAL),
+        update_interval=DEVICE_UPDATE_INTERVAL,
+    )
+    firmware_coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=f"reolink.{host.api.nvr_name}.firmware",
+        update_method=async_check_firmware_update,
+        update_interval=FIRMWARE_UPDATE_INTERVAL,
     )
     # Fetch initial data so we have data when entities subscribe
     try:
-        await coordinator_device_config_update.async_config_entry_first_refresh()
+        await device_coordinator.async_config_entry_first_refresh()
+        await firmware_coordinator.async_config_entry_first_refresh()
```

Comment:
We can use `asyncio.gather` here to do this in parallel.

Reply:
Thanks for the suggestion, done